content/2020-09-30.md

```diff
@@ -0,0 +1,202 @@
+
+## Code Summary
+
+```
+// ==== build.rs ====
+// We manually run the 'cc' compiler and 'ar' archiver to convert our complex
+// C libraries into a simpler, merged library that is compatible with Cargo's limited
+// linker abilities.  Then we tell Cargo about the lib with the "cargo:rustc-link-..." prints.
+//
+// NOTE: The stdout of this script gets stored at target/debug/build/$name-$hash/output.
+
+use std::path::PathBuf;
+
+fn run<S:AsRef<str>>(cmd_str:S) -> Result<(), std::process::ExitStatus> {
+    // Split 'cmd_str' by spaces and create a Command object:
+    let mut pieces = cmd_str.as_ref().split(' ');
+    let mut cmd = std::process::Command::new(pieces.next().unwrap());
+    for piece in pieces {
+        if piece.len()==0 { continue; }
+        cmd.arg(piece);
+    }
+
+    // Run the Command and report the result:
+    let status = cmd.status().unwrap();
+    if status.success() {
+        Ok(())
+    } else {
+        Err(status)
+    }
+}
+
+fn main() {
+    // Helpful for debugging and understanding our context:
+    assert!(std::process::Command::new("env").status().unwrap().success());
+
+    let proj_dir : PathBuf = std::env::var("CARGO_MANIFEST_DIR").unwrap().into();
+    let out_dir  : PathBuf = std::env::var("OUT_DIR").unwrap().into();
+    let lib_dir            = proj_dir.join("x64_lsb-11.17.1.0_v6");
+
+    run(format!("cc -I{lib_dir} -c {proj_dir}/src/c_api.c -o {out_dir}/c_api.o",
+                proj_dir=proj_dir.to_str().unwrap(),
+                lib_dir=lib_dir.to_str().unwrap(),
+                out_dir=out_dir.to_str().unwrap(),
+               )).unwrap();
+
+    run(format!("cc -I{lib_dir} -c {lib_dir}/x64_lsb/lm_new.c -o {out_dir}/lm_new.o",
+                lib_dir=lib_dir.to_str().unwrap(),
+                out_dir=out_dir.to_str().unwrap(),
+               )).unwrap();
+
+    // Note that we are using a "thin" archive to bundle all libs together because Rust's linker controls are extremely limited and unable to handle this kind of circular-dependency + redundant-definitions structure:
+    run(format!("ar crT {out_dir}/libflexbundle.a {out_dir}/c_api.o {out_dir}/lm_new.o {lib_dir}/x64_lsb/liblmgr_pic.a {lib_dir}/x64_lsb/libcrvs_pic.a {lib_dir}/x64_lsb/libsb_pic.a {lib_dir}/x64_lsb/libFNPload_pic.a {lib_dir}/x64_lsb/liblmgr_dongle_pic.a",
+                lib_dir=lib_dir.to_str().unwrap(),
+                out_dir=out_dir.to_str().unwrap(),
+               )).unwrap();
+
+    println!("cargo:rustc-link-search={}", out_dir.to_str().unwrap());
+    println!("cargo:rustc-link-lib=flexbundle");
+}
+```
+
+# An Async Rocket Server with OAuth, Cookies, Diesel, and Templates via [@mysteriouspants](https://github.com/mysteriouspants)
+
+Repo: [https://github.com/mysteriouspants/uDevGames.com](https://github.com/mysteriouspants/uDevGames.com)
```

Comment:
I just transferred this to the iDevGames organization, so the URL is now https://github.com/iDevGames/uDevGames.com (Sorry, I had hoped to get that done *before* the meetup, but wasn't quite fast enough).